### ISSUE TITLE
Plan to remove tf.py_func with actual ops

### ISSUE THREAD

**Issue by KleinYuan** [charlesq34/frustum-pointnets#24]
Thanks for open sourcing the code and I can reproduce the results, even making a light-weight [server](https://github.com/Dark-Rinnegan/frustum-pointnets/tree/app/app) with pre-trained checkpoints.

However, by looking at the graph on tensorboard and the [codes](https://github.com/charlesq34/frustum-pointnets/search?q=py_func&unscoped_q=py_func), it seems you used two [`tf.py_func`](https://www.tensorflow.org/api_docs/python/tf/py_func), which made it difficult to get a frozen graph.

Any plan to replace those two `py_func` with actual ops?

**Comment by charlesq34** [comment]
Hi @KleinYuan 

Great effort in building the server!

The py_func for `compute_box3d_iou` shouldn't matter for inference because it is only used for computing the loss. There has been some third-party effort in rewriting the `mask_to_indices` with TF symbolic operators -- I will see if I can get a permission to share it here.

Best,
Charles

**Comment by DanielHfnr** [comment]
Hi @charlesq34, 

I also ran into a situation where I need the mask_to_indices py_func replaced with actual ops. 
Is it possible to share the code?

Best,
Daniel

**Comment by charlesq34** [comment]
One way to work around is to directly element-wise multiply the mask with the points. It increases the memory usage a bit, but won't affect performance.
It should also be doable to replace it with TF ops but there is no plan to update it any time soon.
Hope it helps.
